Build issue with: error: ‘xdp_portal_create_remote_desktop_session_full’ was not declared in this scope

### What happened?

With build options in build_env.sh:
`B_CMAKE_FLAGS="-DINPUTLEAP_BUILD_LIBEI=ON -DCMAKE_INSTALL_PREFIX=/opt"`

I get this build error resulting from attempted building:
input-leap/src/lib/platform/PortalRemoteDesktop.cpp:164:5: error: ‘xdp_portal_create_remote_desktop_session_full’ was not declared in this scope; did you mean ‘xdp_portal_create_remote_desktop_session_finish’?

(Full error provided bellow)

### Version

From Git HEAD or commit (specify below)

### Git commit hash (if applicable)

_No response_

### If applicable, where did you install Input Leap from?

Compiling from source, current HEAD.


### What OSes are you seeing the problem on? (Check all that apply)

Linux

### If applicable, are you using Wayland or X11?

N/A

### What OS versions are you using?

openSUSE Tumbleweed 20240607

### Relevant log output

```shell
[ 66%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/EiScreen.cpp.o
[ 66%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/PortalInputCapture.cpp.o
[ 67%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/PortalRemoteDesktop.cpp.o
[ 67%] Linking CXX executable ../../bin/guiunittests
/home/psi-jack/src/other/input-leap/src/lib/platform/PortalRemoteDesktop.cpp: In member function ‘gboolean inputleap::PortalRemoteDesktop::init_remote_desktop_session()’:
/home/psi-jack/src/other/input-leap/src/lib/platform/PortalRemoteDesktop.cpp:164:5: error: ‘xdp_portal_create_remote_desktop_session_full’ was not declared in this scope; did you mean ‘xdp_portal_create_remote_desktop_session_finish’?
  164 |     xdp_portal_create_remote_desktop_session_full(
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |     xdp_portal_create_remote_desktop_session_finish
gmake[2]: *** [src/lib/platform/CMakeFiles/platform.dir/build.make:370: src/lib/platform/CMakeFiles/platform.dir/PortalRemoteDesktop.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
[ 67%] Built target guiunittests
[ 67%] Linking CXX executable ../../bin/input-leap
gmake[1]: *** [CMakeFiles/Makefile2:669: src/lib/platform/CMakeFiles/platform.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 84%] Built target input-leap
gmake: *** [Makefile:166: all] Error 2
```


### Any other information

_No response_

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -184,14 +184,15 @@ if (UNIX)
             include_directories(${LIBEI_INCLUDE_DIRS} ${LIBXKBCOMMON_INCLUDE_DIRS} ${GLIB2_INCLUDE_DIRS} ${LIBPORTAL_INCLUDE_DIRS} ${LIBM_INCLUDE_DIRS})
             list(APPEND libs ${LIBEI_LINK_LIBRARIES} ${LIBXKBCOMMON_LINK_LIBRARIES} ${GLIB2_LINK_LIBRARIES} ${LIBPORTAL_LINK_LIBRARIES} ${LIBM_LIBRARIES})
 
-            # no libportal release has xdp_session_connect_to_eis, so let's check for that
-            # and enable the hackaround in the code
+            # libportal 0.7 has xdp_session_connect_to_eis but it doesn't have remote desktop session restore or
+            # the inputcapture code, so let's check for explicit functions that bits depending on what we have
             include(CMakePushCheckState)
             include(CheckCXXSourceCompiles)
             cmake_push_check_state(RESET)
             set(CMAKE_REQUIRED_INCLUDES "${CMAKE_REQUIRED_INCLUDES};${LIBPORTAL_INCLUDE_DIRS};${GLIB2_INCLUDE_DIRS}")
             set(CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_LIBRARIES};${LIBPORTAL_LINK_LIBRARIES};${GLIB2_LINK_LIBRARIES}")
             check_symbol_exists(xdp_session_connect_to_eis "libportal/portal.h" HAVE_LIBPORTAL_SESSION_CONNECT_TO_EIS)
+            check_symbol_exists(xdp_portal_create_remote_desktop_session_full "libportal/portal.h" HAVE_LIBPORTAL_CREATE_REMOTE_DESKTOP_SESSION_FULL)
             check_symbol_exists(xdp_input_capture_session_connect_to_eis "libportal/inputcapture.h" HAVE_LIBPORTAL_INPUTCAPTURE)
             # check_symbol_exists can’t check for enum values
             check_cxx_source_compiles("#include <libportal/portal.h>

--- a/res/config.h.in
+++ b/res/config.h.in
@@ -31,6 +31,9 @@
 /* Define if libportal has xdp_session_connect_to_eis */
 #cmakedefine HAVE_LIBPORTAL_SESSION_CONNECT_TO_EIS ${HAVE_LIBPORTAL_SESSION_CONNECT_TO_EIS}
 
+/* Define if libportal has xdp_portal_create_remote_desktop_session_full */
+#cmakedefine HAVE_LIBPORTAL_CREATE_REMOTE_DESKTOP_SESSION_FULL ${HAVE_LIBPORTAL_CREATE_REMOTE_DESKTOP_SESSION_FULL}
+
 /* Define if libportal has inputcapture support */
 #cmakedefine HAVE_LIBPORTAL_INPUTCAPTURE ${HAVE_LIBPORTAL_INPUTCAPTURE}
 

--- a/src/lib/platform/PortalRemoteDesktop.cpp
+++ b/src/lib/platform/PortalRemoteDesktop.cpp
@@ -158,6 +158,26 @@ void PortalRemoteDesktop::cb_init_remote_desktop_session(GObject* object, GAsync
                       this);
 }
 
+#if !defined(HAVE_LIBPORTAL_CREATE_REMOTE_DESKTOP_SESSION_FULL)
+static inline void
+xdp_portal_create_remote_desktop_session_full(XdpPortal              *portal,
+                                              XdpDeviceType           devices,
+                                              XdpOutputType           outputs,
+                                              XdpRemoteDesktopFlags   flags,
+                                              XdpCursorMode           cursor_mode,
+                                              XdpPersistMode          _unused1,
+                                              const char             *_unused2,
+                                              GCancellable           *cancellable,
+                                              GAsyncReadyCallback     callback,
+                                              gpointer                data)
+{
+    xdp_portal_create_remote_desktop_session(portal, devices, outputs,
+                                             flags, cursor_mode, cancellable,
+                                             callback, data);
+}
+#endif
+
+
 gboolean PortalRemoteDesktop::init_remote_desktop_session()
 {
     LOG_DEBUG("Setting up the RemoteDesktop session with restore token %s", session_restore_token_);

